### ISSUE TITLE
MediaConnect DMS crashes dLeyna

### ISSUE THREAD

**Issue by markdryan** [01org/dleyna-server#49]
It does this by sending a totally invalid response to a UPnP Search query.

```
POST /photolibrary/contentDirectory/control HTTP/1.1
Host: 192.168.1.134:9080
SOAPAction: "urn:schemas-upnp-org:service:ContentDirectory:1#Search"
Accept-Encoding: gzip
Content-Type: text/xml; charset="utf-8"
User-Agent: dLeyna/0.0.2 GUPnP/0.20.1 DLNADOC/1.50
Connection: Keep-Alive
Content-Length: 672

<?xml version="1.0"?><s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><s:Body><u:Search xmlns:u="urn:schemas-upnp-org:service:ContentDirectory:1"><RequestedCount>50</RequestedCount><StartingIndex>0</StartingIndex><SortCriteria>-dc:title</SortCriteria><SearchCriteria> ( upnp:class derivedfrom &quot;object.item.imageItem&quot;  or upnp:class derivedfrom &quot;object.item.videoItem&quot;  or upnp:class derivedfrom &quot;object.item.audioItem&quot;  ) and ( @refID exists false )</SearchCriteria><Filter>res,dc:date,upnp:artist</Filter><ContainerID>0</ContainerID></u:Search></s:Body></s:Envelope>HTTP/1.1 200 OK
Accept-Ranges: bytes
Content-Length: 249
Cache-Control: max-age=1800
Server: iPhoneOS/6.1.3, UPnP/1.0, MediaConnect/2.0
Content-Type: text/xml; charset="utf-8"
Date: ven., 29 mars 2013 13:11:31 GMT

<s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><s:Body><u:SearchResponse xmlns:u="urn:schemas-upnp-org:service:ContentDirectory:1"></u:SearchResponse></s:Body></s:Envelope>
```

Note that it returns success but it doesn't fill in any of the output parameters.

This causes the following crash in dLeyna-server

```
#0  __strlen_sse2_pminub () at ../sysdeps/x86_64/multiarch/strlen-sse2-pminub.S:39
#1  0x00007ffff5e998b7 in gupnp_didl_lite_parser_parse_didl_recursive (parser=0x676ce0, didl=0x0, recursive=0, error=0x7fffffffe100) at gupnp-didl-lite-parser.c:237
#2  0x00007ffff76bfaf7 in prv_search_cb (proxy=0x64cbf0, action=0x8b90d0, user_data=0x68c480) at device.c:2600
```

The reason that we are crashing is that gupnp returns true when we call gupnp_service_proxy_end_action.  We assume that this means that all the output parameters have been correctly filled and so we do not test them.  Unfortunately, as they are not filled the result parameter is NULL.  dLeyna assumes it is not NULL and merrily deferences it.  it's not clear to me where this should be fixed, GUPnP or dLeyna.  I'm thinking GUPnP as it knows which output parameters are missing and it can fail the request.  dLeyna can detect when string parameters are not initialised but it can't really tell when ints are not correctly initialised.

Basically, I reproduced this by selecting a MediaConnect server that was sharing no media.
